org.mozilla.Thunderbird.json

```diff
@@ -0,0 +1,92 @@
+{
+    "app-id": "org.mozilla.Thunderbird",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "thunderbird",
+    "rename-icon": "thunderbird",
+    "finish-args": [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* OpenGL access */
+        "--device=dri",
+        /* Needs to talk to the network: */
+        "--share=network",
+        /* Play sounds */
+        "--socket=pulseaudio",
+        /* Access to thunderbird configuration and cache files */
+        "--filesystem=~/.cache/thunderbird", "--filesystem=~/.thunderbird",
+        /* Access to home directory (for attachments) */
+        "--filesystem=home",
+        /* Needed for dconf to work */
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g",
+        "env": {
+            "V": "1"
+        }
+    },
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/pkgconfig",
+        "/share/aclocal",
+        "/man",
+        "/share/man",
+        "*.la", "*.a"
+    ],
+    "modules": [
+        {
+            "name": "autoconf-2.13",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnu.org/gnu/autoconf/autoconf-2.13.tar.gz",
+                    "sha256": "f0611136bee505811e9ca11ca7ac188ef5323a8e2ef19cffd3edb3cf08fd791e"
+                }
+            ],
+            "post-install": [
+	        "ln -s /app/bin/autoconf /app/bin/autoconf-2.13"
+            ]
+        },
+	{
+	    "name": "icu",
+	    "sources": [
+	       {
+		    "type": "archive",
+	            "url":"http://download.icu-project.org/files/icu4c/58.2/icu4c-58_2-src.tgz",
+	            "sha256":"2b0a4410153a9b20de0e20c7d8b66049a72aef244b53683d0d7521371683da0c"
+	       }
+	    ],
+	   "subdir": "source"
+	},
+        {
+            "name": "thunderbird",
+            "no-autogen": true,
```

Comment:
Please use the `simple` buildsystem and `build-commands` instead.

Reply:
How do I do this? Sorry, I'm new to flatpak.

Reply:
Instead of setting `rm-configure` and `no-autogen`, set `buildsystem` to `simple` and put commands from Makefile into `build-commands` (it's an array of strings).

Reply:
Still rather unsure where `build-commands` is supposed to go in the JSON file, and the structure of the items I'd be putting inside it.

Reply:
Do you mean something structured like?:

```json
        {
            "name": "thunderbird",
            "buildsystem": "smiple",
            "build-commands": [],
```

Reply:
@Barthalion is right, you should use the simple build system here, but there is a typo (`smiple` -> `simple`) in the snippet above, and also you'll need to provide actual commands to build in the `build-commands` section, of course.

Reply:
FWIW, the only reason I didn't use the simple build system myself is that it didn't exist at the time, it was added later: https://github.com/flatpak/flatpak/pull/516

Reply:
```JSON
"buildsystem": "smiple",
            "build-options":{
                "cflags": "-fno-delete-null-pointer-checks -fno-lifetime-dse -fno-schedule-insns2",
                "cxxflags": "-fno-delete-null-pointer-checks -fno-lifetime-dse -fno-schedule-insns2",
                "env":{
                    "VERSION": "52.4.0"
                }
            },
            "build-commands": [
                "all: make -f client.mk",
                "instal:\n make -f client.mk install INSTALL_SDK= \n mkdir -p /app/share/icons/hicolor/16x16/apps \n cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default16.png /app/share/icons/hicolor/16x16/apps/thunderbird.png \n mkdir -p /app/share/icons/hicolor/22x22/apps \n cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default22.png /app/share/icons/hicolor/22x22/apps/thunderbird.png \n mkdir -p /app/share/icons/hicolor/24x24/apps \n cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default24.png /app/share/icons/hicolor/24x24/apps/thunderbird.png \n mkdir -p /app/share/icons/hicolor/32x32/apps \n cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default32.png /app/share/icons/hicolor/32x32/apps/thunderbird.png \n mkdir -p /app/share/icons/hicolor/48x48/apps cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default48.png /app/share/icons/hicolor/48x48/apps/thunderbird.png \n mkdir -p /app/share/icons/hicolor/256x256/apps \n cp /app/lib/thunderbird-$(VERSION)/chrome/icons/default/default256.png /app/share/icons/hicolor/256x256/apps/thunderbird.png \n mkdir -p /app/share/applications \n cp org.mozilla.Thunderbird.desktop /app/share/applications"
            ],
```

Something like this?

Reply:
Hrmmm...after doing this I get a build error:

Reply:
I have never used the `simple` build system before, so perhaps not the best person to comment on this, but it looks to me like you should at least split the second line under `build-commands` into several lines, one for each actual command you're putting there (instead of having everything in one line separated by `\n`).

But as I said, I'm not the most qualified one to comment on this build system.

Reply:
I can't because JSON doesn't seem to support multiline @mariospr ?

Reply:
Good idea

Reply:
I finally wrestled with it and "fixed" it.